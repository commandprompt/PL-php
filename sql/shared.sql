--
-- $_SHARED test
--

CREATE OR REPLACE FUNCTION php_set_shared (text, text) RETURNS text AS $$
  global $_SHARED;
  $_SHARED[$args[0]] = $args[1];
  return 'ok';
$$ LANGUAGE plphp;

CREATE OR REPLACE FUNCTION php_get_shared (text) RETURNS text AS $$
  global $_SHARED;
  return $_SHARED[$args[0]];
$$ LANGUAGE plphp;

SELECT php_set_shared('first', 'hello plphp');
SELECT php_get_shared('first');

SELECT php_get_shared('second');

SELECT php_set_shared('third', 'hello again!');
SELECT php_set_shared('third', 'say goodbye');
SELECT php_get_shared('third');

CREATE OR REPLACE FUNCTION php_get_shared_ary (text) RETURNS text[] AS $$
  global $_SHARED;
  return $_SHARED[$args[0]];
$$ LANGUAGE plphp;

SELECT php_set_shared('fourth', $${'hip', 'hooray'}$$);
SELECT php_get_shared('fourth');
SELECT php_get_shared_ary('fourth');
SELECT (php_get_shared_ary('fourth'))[1];
SELECT (php_get_shared_ary('fourth'))[1];
SELECT (php_get_shared_ary('fourth'))[2];
