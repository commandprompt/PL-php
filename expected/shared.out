--
-- $_SHARED test
--
CREATE OR REPLACE FUNCTION php_set_shared (text, text) RETURNS text AS $$
  global $_SHARED;
  $_SHARED[$args[0]] = $args[1];
  return 'ok';
$$ LANGUAGE plphp;
CREATE OR REPLACE FUNCTION php_get_shared (text) RETURNS text AS $$
  global $_SHARED;
  return $_SHARED[$args[0]];
$$ LANGUAGE plphp;
SELECT php_set_shared('first', 'hello plphp');
 php_set_shared 
----------------
 ok
(1 row)

SELECT php_get_shared('first');
 php_get_shared 
----------------
 hello plphp
(1 row)

SELECT php_get_shared('second');
NOTICE:  plphp: Undefined index:  second
 php_get_shared 
----------------
 
(1 row)

SELECT php_set_shared('third', 'hello again!');
 php_set_shared 
----------------
 ok
(1 row)

SELECT php_set_shared('third', 'say goodbye');
 php_set_shared 
----------------
 ok
(1 row)

SELECT php_get_shared('third');
 php_get_shared 
----------------
 say goodbye
(1 row)

CREATE OR REPLACE FUNCTION php_get_shared_ary (text) RETURNS text[] AS $$
  global $_SHARED;
  return $_SHARED[$args[0]];
$$ LANGUAGE plphp;
SELECT php_set_shared('fourth', $${'hip', 'hooray'}$$);
 php_set_shared 
----------------
 ok
(1 row)

SELECT php_get_shared('fourth');
ERROR:  this plphp function cannot return arrays
SELECT php_get_shared_ary('fourth');
 php_get_shared_ary 
--------------------
 {hip,hooray}
(1 row)

SELECT (php_get_shared_ary('fourth'))[1];
 php_get_shared_ary 
--------------------
 hip
(1 row)

SELECT (php_get_shared_ary('fourth'))[1];
 php_get_shared_ary 
--------------------
 hip
(1 row)

SELECT (php_get_shared_ary('fourth'))[2];
 php_get_shared_ary 
--------------------
 hooray
(1 row)

